### ISSUE TITLE
[ BUG REPORT ] - Previously configured outfit may appear briefly during a fade to black transition

### ISSUE THREAD

**Issue by RZed01** [Just-Natsuki-Team/NatsukiModDev#849]
# Overview:
If you change Natsuki's outfit, let her slip into certain idle animations (such as reading books), and then interrupt her, there is a rare chance she may briefly wear the outfit that she wore before you changed it when the fade to black transition triggers. This is only temporary, and she reverts to the chosen outfit after the fade to black.

## Replication steps:
Since this bug only happens with certain animations that involve transitions (and it does not even happen consistently), it cannot really be replicated consistently on my end. How this bug mainly happens is that you launch the game with custom outfits already unlocked, and then change her outfit. Then, let her do an idle animation where she brings out either a book or her headphones. After interrupting her, and letting her put the book/headphones down or away, this bug has a chance of happening during the fade to black (i.e. 1-2 seconds). After that, she reverts to wearing the outfit you chose.

## Additional information:
Observed on Windows 10 22H2 build 19045.3693, on two different occasions- once on 11/28/2023, and once on 11/29/2023. This is a very small issue that does not actually affect anything (even the logs say nothing), so assigning the "Low Priority" tag is recommended. It is still a bug, so I am only including this here for polishing reasons. If the solution involves removing said idle animations, please close this issue.

Update 12/1: I've managed to narrow down the bug somewhat. It is not necessarily the launch outfit that this bug may happen with, but the _last outfit config_ that was stored in memory, apparently. This bug also does not only happen with idle animations, but potentially any animation that involves a fade to black. It is like the RAM buffer was not wiped after changing the outfit, so the previous outfit is stored in memory just in case it needs to be accessed again.

**Comment by RZed01** [comment]
I do have to say though, I love those idle animations. They are very immersive and make this mod quite unique to be honest.

**Comment by RZed01** [comment]
Here's something that may help. I encountered this bug three times today, all happening during the fade to black transition, but two of them were not from an idle animation. (The first time involved Natsuki's headphones briefly appearing on her head during a fade to black- her headphones having already appeared in an idle two fade to black transitions earlier) Based on the patterns that I have seen for this bug, I believe the culprit is the way this mod stores outfits- that previously configured outfits probably continue to be stored in memory or in a file and not deleted/overwritten after every change, despite the fact there is no undo function for outfit changes. It does not even have to be an outfit I set myself- Natsuki putting on her headphones counts as an outfit change that is then stored.
